File: Python/LinkedStack.py
class LinkedStack:

    class _Node:
        __slots__ = '_element', '_next'

        def __init__(self, element, next):
            self._element = element
            self._next = next

    def __init__(self):
        self._head = None
        self._size = 0

    def __len__(self):
        return self._size

    def is_empty(self):
        return self._size == 0;

    def push(self, e):
        """Add element e to the top of the stack."""
        self._head = self._Node(e, self._head)
        self._size += 1

    def top(self):
        if self.is_empty():
            print('Stack is empty.')
            return
        return self._head._element

    def pop(self):
        if self.is_empty():
            print('Stack is empty.')
            return
        answer = self._head._element
        self._head = self._head._next
        self._size -= 1
        return answer


if __name__ == '__main__':
    myStack = LinkedStack()
    myStack.push(1)
    myStack.push(2)
    print(myStack.top())
    myStack.pop()
    print(myStack.top())
    print(myStack.__len__)

File: CSharp/LinkedStack.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CSharp
{
    public class Node<T>
    {
        public Node(T element, Node<T> next)
        {
            Element = element;
            Next = next;
        }

        public T Element { get; set; }
        
        public Node<T> Next { get; set; }
    }

    public class LinkedStack<T>
    {
        public LinkedStack()
        {
            Head = new Node<T>(default, null);
            Size = 0;
        }

        public Node<T> Head;

        public int Size { get; set; }

        public int Length
        {
            get { return Size; }
        }

        public bool IsEmpty
        {
            get { return Size.Equals(0); }
        }

        public void Push(T element)
        {
            // Add element to top of stack.
            Head = new Node<T>(element, Head);
            Size += 1;
        }

        public T Top()
        {
            // Returns but does not remove the element at the top of the stack.
            if (this.IsEmpty)
            {
                throw new Exception("The list is empty!");
            }
            return Head.Element;
        }

        public T Pop()
        {
            // Remove and return the element from the top of the stack (LIFO).
            if (this.IsEmpty)
            {
                throw new Exception("The list is empty!");
            }
            var result = Head.Element;
            Head = Head.Next;
            Size -= 1;
            return result;
        }
    }
}
